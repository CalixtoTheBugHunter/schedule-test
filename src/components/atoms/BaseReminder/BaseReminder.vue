<template>
    <div class="BaseReminder" :style="BaseReminderStyles" @click="onClick">
        {{ title }}
    </div>
</template>

<script>
export default {
    name: 'BaseReminder',
    props: {
        title: {
            type: String,
            required: true
        },
        color: {
            type: String,
            default: '#000'
        }
    },
    emits: ['click'],
    computed: {
        BaseReminderStyles() {
            return `background: ${this.color}`
        }
    },
    methods: {
        onClick() {
            this.$emit('click')
        }
    }
}
</script>

<style lang="scss" scoped>
.BaseReminder {
    color: $color-white;
    cursor: pointer;
}
</style>