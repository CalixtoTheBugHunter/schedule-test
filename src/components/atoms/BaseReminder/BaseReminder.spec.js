/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import BaseReminder from './BaseReminder'

describe('BaseReminder', () => {
  const WRAPPER = shallowMount(BaseReminder)

  it('Component exists', () => {
    expect(WRAPPER.exists()).toBe(true)
  })

  it('Component not changed', () => {
    expect(WRAPPER.html()).toMatchSnapshot()
  })
})