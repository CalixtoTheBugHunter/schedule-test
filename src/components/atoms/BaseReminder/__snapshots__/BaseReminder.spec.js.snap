// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseReminder Component not changed 1`] = `<div class="BaseReminder"></div>`;
